src/mail.sh

```diff
@@ -0,0 +1,396 @@
+
+  cmd_manager "$flag" "$cmd"
+
+  if [[ "$?" == 0 ]] && [[ "$flag" != 'TEST_MODE' ]]; then
+    success "$option was successfully set to '${options_values[$option]}'"
+  fi
+}
+
+# This function gets all the currently set values for the configurations used
+# by this script and writes them to the global variable set_confs
+#
+# @cmd_scope:  The scope being imposed on the command
+# @set_values: The values of the option that are already set
+# @option:     The option name
+# @value:      The value to update the option to
+# @scope:      The scope of the given value
+#
+function get_configs()
+{
+  local cmd_scope=${options_values['CMD_SCOPE']}
+  local -a set_values
+  local option
+  local value
+  local scope
+  local i=0
+
+  IFS=$'\t\n' read -r -a set_values -d '' <<< "$(get_git_config_regex 'user|sendemail\.smtp' "$cmd_scope")"
+
+  while [[ -n "${set_values[$((i + 1))]}" ]]; do
+    scope="${set_values[$i]}"
+    # These separate the option from the value as they come as a single value
+    # this removes the suffix of the string
+    option="${set_values[$((i + 1))]%% *}"
+    # this removes the prefix of the string
+    value="${set_values[$((i + 1))]#* }"
+
+    if [[ "${essential_config_options[*]}" =~ "$option".* ]]; then
+      set_confs["${scope}_$option"]="$value"
+    fi
+    if [[ "${optional_config_options[*]}" =~ "$option".* ]]; then
+      set_confs["${scope}_$option"]="$value"
+    fi
+    i=$((i + 2))
+  done
+}
+
+# This function checks that the required and recommended options to use
+# git send-email have been set. It does not validate the options values, only
+# that a value has been set.
+#
+# @missing_conf:     The options that have not been set
+# @missing_rec_conf: The recommended options that have not been set
+# @cmd_scope:            Limit the search scope
+# @set_confs:        The relevant options that have already been set
+#
+# Returns:
+# Returns 22 if missing any required configuration; 0 otherwise
+function mail_verify()
+{
+  local count=0
+  local count_opt=0
+  local -a missing_conf
+  local -a missing_opt_conf
+  local cmd_scope=${options_values['CMD_SCOPE']}
+
+  for config in {'local','global'}'_sendemail.smtpserver'; do
+    if [[ -d "${set_confs[$config]}" ]]; then
+      warning 'It appears you are using a local smtpserver with custom configurations.'
+      warning "Unfortunately we can't verify these configurations yet."
+      warning "  Current value is: ${set_confs[$config]}"
+
+      return 0
+    fi
+  done
+
+  for config in "${optional_config_options[@]}"; do
+    if [[ ! "${!set_confs[*]} " =~ (local|global)_"$config"[[:space:]] ]]; then
+      missing_opt_conf[$count_opt]="$config"
+      count_opt=$((count_opt + 1))
+    fi
+  done
+
+  for config in "${essential_config_options[@]}"; do
+    if [[ ! "${!set_confs[*]} " =~ (local|global)_"$config"[[:space:]] ]]; then
+      missing_conf[$count]="$config"
+      count=$((count + 1))
+    fi
+  done
+
+  if [[ $count -gt 0 ]]; then
```

Comment:
I think it's because the `-gt` operator makes the left side be interpreted as an arithmetic expression, and therefore expects any variable to just be a number, which wouldn't cause any expansion problems. But I will add the quotes anyway.